File: packages/csharp/HtmlToMarkdown/Metadata/ExtractTable.cs
using System.Text.Json.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// A single table extracted via Extract(), containing structured cell data and Markdown.
/// </summary>
public record ExtractTable
{
    /// <summary>
    /// The structured table grid with row/column/cell data.
    /// </summary>
    [JsonPropertyName("grid")]
    public TableGrid Grid { get; init; } = new();

    /// <summary>
    /// The rendered Markdown representation of the table.
    /// </summary>
    [JsonPropertyName("markdown")]
    public string Markdown { get; init; } = string.Empty;
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/ExtractTable.java
package dev.kreuzberg.htmltomarkdown;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A single table extracted via {@code extract()}, containing structured cell data and markdown.
 *
 * @param grid the structured table grid with row/column/cell data
 * @param markdown the rendered Markdown representation of the table
 */
public record ExtractTable(
    @JsonProperty("grid") TableGrid grid, @JsonProperty("markdown") String markdown) { }
